infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -56,16 +60,50 @@ def run_shell_command(command, env_vars):
 ###################
 
 
-def make_backup_fpath():
-    return BACKUP_DIR / datetime.now().strftime(FILENAME_DATETIME_FORMAT)
+def make_object_name():
+    return datetime.now().strftime(FILENAME_DATETIME_FORMAT)
 
 
-def backup_database(db_backup_gzip_fpath):
+def make_backup_fpath(object_name):
+    return BACKUP_DIR / object_name
+
+
+def create_s3_client():
+    """Creates and returns an AWS S3 client for uploading backup to the cloud."""
+    sess = boto3.session.Session(
+        aws_access_key_id=BACKUP_DB__AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=BACKUP_DB__AWS_SECRET_ACCESS_KEY,
+    )
+    return sess.client("s3")
+
+
+def create_s3_session():
+    """Creates and returns an AWS session for listing and downloading backups in S3."""
+    sess = boto3.session.Session(
+        aws_access_key_id=BACKUP_DB__AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=BACKUP_DB__AWS_SECRET_ACCESS_KEY,
+    )
+    return sess
+
+
+def upload_backup_to_s3(s3_client, backup_fpath, backup_bucket_name, backup_object_name):
+    """Uploads the backup_fpath file to AWS S3."""
```

Comment:
You're probably on this, but parameters should be documented with the `:param <name>: <description>` format. This way we can auto-generate documentation about our code!

Reply:
I have done this using the Google format which I prefer for the functions with more than one parameter. should I do this for all of them?

Reply:
specified params everywhere for custom functions and added full docstrings to all functions